### DISCUSSION TITLE
How can I get the (1) latest tweet for a specific keyword with Tweepy?

### DISCUSSION THREAD

**Discussion by 621625** [tweepy/tweepy#1637] [Questions] [answered]
How can I get the (1) latest tweet for a specific keyword with Tweepy?

For example, I want the latest one tweet about the keyword "socks" including username of the tweeter. 

How would the code look like? Please be specific.

**Comment by Harmon758** [comment, marked as answer]
You can use Twitter's [standard search API](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/search/api-reference/get-search-tweets) with [`API.search`](https://tweepy.readthedocs.io/en/v3.10.0/api.html#API.search) and set the `count` parameter to `1` and the `q` parameter to your keyword, e.g. `"socks"`.

Alternatively, you can also use Twitter's [premium search APIs](https://developer.twitter.com/en/docs/twitter-api/premium/search-api/api-reference/premium-search) with [`API.search_30_day`](https://tweepy.readthedocs.io/en/v3.10.0/api.html#API.search_30_day) and/or [`API.search_full_archive`](https://tweepy.readthedocs.io/en/v3.10.0/api.html#API.search_full_archive); set the `query` parameter to your keyword, e.g. `"socks"`; set the `maxResults` parameter to the minimum `10`; and only use the first of the results returned.

You can then use the `user` attribute of the Status/[Tweet object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/tweet) to retrieve the [User object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/user) representing the user who posted the Tweet and the `screen_name` attribute of that User object for their username.
